cupyx/scipy/sparse/csr.py

```diff
@@ -348,6 +347,96 @@ def transpose(self, axes=None, copy=False):
         trans.has_canonical_format = self.has_canonical_format
         return trans
 
+    def getrow(self, i):
+        """Returns a copy of row i of the matrix, as a (1 x n)
+        CSR matrix (row vector).
+
+        Args:
+            i (integer): Row
+
+        Returns:
+            cupyx.scipy.sparse.csr_matrix: Sparse matrix with single row
+        """
+        M, N = self.shape
+        i = int(i)
+        if i < 0:
+            i += M
+        if i < 0 or i >= M:
+            raise IndexError('index (%d) out of range' % i)
+        indptr, indices, data = index._get_csr_submatrix(
+            self.indptr, self.indices, self.data, i, i + 1, 0, N)
+        return csr_matrix((data, indices, indptr), shape=(1, N),
+                          dtype=self.dtype, copy=False)
+
+    def getcol(self, i):
+        """Returns a copy of column i of the matrix, as a (m x 1)
+        CSR matrix (column vector).
+
+        Args:
+            i (integer): Column
+
+        Returns:
+            cupyx.scipy.sparse.csr_matrix: Sparse matrix with single column
+        """
+        M, N = self.shape
+        i = int(i)
+        if i < 0:
+            i += N
+        if i < 0 or i >= N:
+            raise IndexError('index (%d) out of range' % i)
+        indptr, indices, data = index._get_csr_submatrix(
+            self.indptr, self.indices, self.data, 0, M, i, i + 1)
+        return csr_matrix((data, indices, indptr), shape=(M, 1),
+                          dtype=self.dtype, copy=False)
+
+    def _get_intXarray(self, row, col):
+        raise NotImplementedError()
+
+    def _get_intXslice(self, row, col):
+        if col.step in (1, None):
+            return self._get_submatrix(row, col, copy=True)
+
+        M, N = self.shape
+        start, stop, stride = col.indices(N)
+
+        ii, jj = self.indptr[row:row+2]
+        row_indices = self.indices[ii:jj]
+        row_data = self.data[ii:jj]
+
+        if stride > 0:
+            ind = (row_indices >= start) & (row_indices < stop)
+        else:
+            ind = (row_indices <= start) & (row_indices > stop)
+
+        if abs(stride) > 1:
+            ind &= (row_indices - start) % stride == 0
+
+        row_indices = (row_indices[ind] - start) // stride
+        row_data = row_data[ind]
+        row_indptr = cupy.array([0, len(row_indices)])
+
+        if stride < 0:
+            row_data = row_data[::-1]
+            row_indices = abs(row_indices[::-1])
+
+        shape = (1, int(cupy.ceil(float(stop - start) / stride)))
```

Comment:
I think `stop`, `start`, and `stride` are all `int` on CPU, and perhaps this works?
```suggestion
        shape = (1, (stop - start + stride - 1) // stride)
```